src/task.ts

```diff
@@ -74,13 +88,7 @@ export default class Task extends EventTarget {
           await this.opts.beforeEach.call(this);
         }
 
-        const taskStart = this.bench.now();
-        if (isAsync) {
-          await this.fn();
-        } else {
-          this.fn();
-        }
-        const taskTime = this.bench.now() - taskStart;
+        const taskTime = isAsync ? await this.runAsync() : this.runSync();
```

Comment:
why this change? doesn't it do exactly same thing but with more code?

Reply:
previously, the condition itself was included in the challenge
for quick functions it makes sense

Reply:
What do you mean? Isn't exactly the same code going to be executed the same amount of times now, just spread differently?

Reply:
In my changes, the calculation is in runAsync/runSync
I mean the execution of the condition is above the time calculation, not in the middle

Reply:
@sirenkovladd I don't see that. Previously it calculated starting point, executed function, calculated taskTime. It does exactly same thing now.

Reply:
https://github.com/tinylibs/tinybench/blob/4d2c04f564586bdb3c8914fee54117a8faa0fc9e/src/task.ts#L77-L83

I see a condition in the calculation of time

Reply:
If you want I can take out the `while` from the `run` in `runAsync`/`run/Sync`

Reply:
I feel that we are talkibg about very different things. Can we hop to discord to discuss in realtime, perhaps?